module-gui/gui/core/FontManager.cpp

```diff
@@ -33,29 +53,31 @@ namespace gui
 
     void FontManager::loadFonts(std::string baseDirectory)
     {
-        fontFolder                         = baseDirectory + "/fonts";
-        std::vector<std::string> fontFiles = getFontsList();
+        fontFolder     = baseDirectory + "/fonts";
+        fontMapFile    = fontFolder + "/fontmap.json";
+        auto fontFiles = getFontsList();
 
-        for (std::string fontName : fontFiles) {
-            loadFont(fontName);
+        for (auto font : fontFiles) {
+            loadFont(font.first, font.second);
         }
     }
 
-    RawFont *FontManager::loadFont(std::string filename)
+    RawFont *FontManager::loadFont(Fonts font, std::string path)
     {
 
-        auto file = std::fopen(filename.c_str(), "rb");
+        auto file = std::fopen(path.c_str(), "rb");
         if (file == nullptr) {
+            LOG_ERROR("Failed to open file: %s", path.c_str());
             return nullptr;
         }
 
-        auto fileSize = std::filesystem::file_size(filename);
+        auto fileSize = std::filesystem::file_size(path);
         std::rewind(file);
 
         char *fontData = new char[fileSize];
```

Comment:
Replace it with `std::unique_ptr<char[]> p(new char[fileSize])` or use plain std::vector. Then you won't need to manually manage memory.